### PR TITLE
Enable SSL termination for the container for local dev

### PR DESCRIPTION
**Pull request by nalshamma**
@bryanlatten Still WIP. I ended up doing two implementations, and prefer the one using CONTAINER_SSL and CONTAINER_PORT. I would remove the SERVER_ENABLE_SSL implementation. 

The tests currently fail if you use --build-args CONTAINER_SSL=true with this problem, although I can run with https on the container: 

$ nginx -t
nginx: [warn] conflicting server name "" on 0.0.0.0:8080, ignored

I have deferred working on alpine while still being reviewed.

### CODE REVIEW

**Comment by nalshamma** [comment]
@bryanlatten Ready for review. SSL is now enabled at "docker run" time. Certs are mapped from local file system. Tested both ubuntu and alpine build & run for SSL.

File: Dockerfile [hunk 1]
```diff
@@ -22,6 +22,9 @@ RUN /bin/bash -e /security_updates.sh && \
     apt-get install -yqq --no-install-recommends \
         nginx-light \
     && \
+    apt-get install -yqq --no-install-recommends \
```
Comment (bryanlatten) [inline Dockerfile]: this installation can be combined right below `nginx-light`, no need for an additional install line

File: Dockerfile-alpine [hunk 2]
```diff
@@ -19,6 +19,9 @@ RUN apk update --no-cache && \
     apk add \
         nginx \
     && \
+    apk add \
```
Comment (bryanlatten) [inline Dockerfile-alpine]: can also be combined with above

File: README.md [hunk 3]
```diff
@@ -7,11 +7,11 @@ Ubuntu used by default, Alpine builds also available tagged as `-alpine`
 
 Provides base OS, patches and stable nginx for quick and easy spinup.
 
-[S6](https://github.com/just-containers/s6-overlay) process supervisor is used for `only` for zombie reaping (as PID 1), boot coordination, and termination signal translation  
+[S6](https://github.com/just-containers/s6-overlay) process supervisor is used for `only` for zombie reaping (as PID 1), boot coordination, and termination signal translation
```
Comment (bryanlatten) [inline README.md]: It looks like your editor clipped the whitespace, which is important in MD files

**Comment by nalshamma** [comment]
@bryanlatten Okay, I made updates per suggestions.

File: container/root/etc/nginx/sites-available/default [hunk 4]
```diff
@@ -1,6 +1,15 @@
 server {
   listen 8080;
 
+  #ssl on;
+  #ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
```
Comment (bryanlatten) [inline container/root/etc/nginx/sites-available/default]: please remove TLSv1 (and possibly 1.1) from the list

Comment (bryanlatten) [reply container/root/etc/nginx/sites-available/default]: https://blog.pcisecuritystandards.org/are-you-ready-for-30-june-2018-sayin-goodbye-to-ssl-early-tls

**Comment by bryanlatten** [comment]
@nalshamma a PR was just merged adding cent to the mix. Can you handle that one too?

**Comment by nalshamma** [comment]
@bryanlatten merge completed

**Comment by bryanlatten** [comment]
Superseded by https://github.com/behance/docker-nginx/pull/74

NOTE: the warning for duplicate server_name is due to your `sed -ig` which duplicated the file

**Pull request closed without merging**